src/restruct_poc/types.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+namespace oturb {
+
+static constexpr std::size_t LieAlgebraComponents = 6;
+static constexpr std::size_t LieGroupComponents = 7;
+static constexpr std::size_t VectorComponents = 3;
+
+static constexpr double Tolerance = 1.e-16;
+
+using LieAlgebraFieldView = Kokkos::View<double* [LieAlgebraComponents]>;
+using LieGroupFieldView = Kokkos::View<double* [LieGroupComponents]>;
+using VectorFieldView = Kokkos::View<double* [VectorComponents]>;
+
+using View1D = Kokkos::View<double*>;
+using View1D_Vector = Kokkos::View<double[VectorComponents]>;
+using View1D_LieAlgebra = Kokkos::View<double[LieAlgebraComponents]>;
+using View1D_LieGroup = Kokkos::View<double[LieGroupComponents]>;
+
+using View_Rot = Kokkos::View<double[3][3]>;
+using View_3x3 = Kokkos::View<double[3][3]>;
+using View_3x4 = Kokkos::View<double[3][4]>;
+using View_6x6 = Kokkos::View<double[6][6]>;
+
+using View_N = Kokkos::View<double*>;
+using View_3 = Kokkos::View<double[3]>;
+using View_Quat = Kokkos::View<double[4]>;
```

Comment:
On a similar note to Line 22, could we call it `using View_Quaternion = Kokkos::View<double[4]>;`?

Reply:
Yes, `View_Quaternion` is a better name and is only used in a couple places.

Reply:
Thanks!

In general, I like the type names to be long, descriptive, and fully spelled out like a word. It improves the reading experience in my opinion, especially for someone who's not a vested developer of the repository.